---
title: "Introduction to Single Cell RNAseq Part 4"
author: "UCD Bioinformatics Core"
output:
    html_document:
      keep_md: TRUE
---

Last Updated: March 23 2021, 5pm

# Part 4: PCA and choice in number of PCS

## Load libraries
```{r load_libraries, warning=FALSE,error=FALSE,message=FALSE}
library(Seurat)
library(ggplot2)
```

## Load the Seurat object
```{r load_rdata, warning=FALSE,error=FALSE,message=FALSE}
load(file="pre_sample_corrected.RData")
#For using all genes
#VariableFeatures(experiment.aggregate) <-rownames(experiment.aggregate)
#newVariableFeatures <- VariableFeatures(experiment.aggregate)[1:1000]
experiment.aggregate

```


## Now doing so for 'real'

ScaleData - Scales and centers genes in the dataset. If variables are provided in vars.to.regress, they are individually regressed against each gene, and the resulting residuals are then scaled and centered unless otherwise specified. Here we regress out cell cycle results S.Score and G2M.Score, percentage mitochondria (percent.mito) and the number of features (nFeature_RNA).

```{r scale_data, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- ScaleData(
  object = experiment.aggregate,
  vars.to.regress = c("batchid", "S.Score", "G2M.Score", "percent.mito", "nFeature_RNA"))
 # vars.to.regress = c("batchid", "S.Score",  "percent.mito", "nFeature_RNA"))
```

## Dimensionality reduction with PCA

Next we perform PCA (principal components analysis) on the scaled data.  

```{r pca_help, warning=FALSE,error=FALSE,message=FALSE, eval=FALSE}
?RunPCA
```

```{r pca, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- RunPCA(object = experiment.aggregate)
#experiment.aggregate <- RunPCA(object = experiment.aggregate, feature=newVariableFeatures)
```

Seurat then provides a number of ways to visualize the PCA results

Visualize PCA loadings
```{r viz_pca, warning=FALSE,error=FALSE,message=FALSE}
#magnitute
VizDimLoadings(experiment.aggregate, dims = 1, ncol = 1) + theme_minimal(base_size = 8)
VizDimLoadings(experiment.aggregate, dims = 2, ncol = 1) + theme_minimal(base_size = 8)
```

Principal components plot
```{r plot_pca, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate, reduction = "pca")
```

Draws a heatmap focusing on a principal component. Both cells and genes are sorted by their principal component scores. Allows for nice visualization of sources of heterogeneity in the dataset.

```{r heatmap_pca, warning=FALSE,error=FALSE,message=FALSE}
DimHeatmap(object = experiment.aggregate, dims = 1:6, cells = 500, balanced = TRUE)

DimHeatmap(object = experiment.aggregate, dims = 7:12, cells = 500, balanced = TRUE)
```

#### Questions

1. Go back to the original data (rerun the load RData section) and then try modifying the ScaleData vars.to.regres, remove some variables, try adding in orig.ident? See how choices effect the pca plot

### Selecting which PCs to use
To overcome the extensive technical noise in any single gene, Seurat clusters cells based on their PCA scores, with each PC essentially representing a metagene that combines information across a correlated gene set. Determining how many PCs to include downstream is therefore an important step.

ElbowPlot plots the standard deviations (or approximate singular values if running PCAFast) of the principle components for easy identification of an elbow in the graph. This elbow often corresponds well with the significant PCs and is much faster to run.  This is the traditional approach to selecting principal components.

```{r elbow, warning=FALSE,error=FALSE,message=FALSE}
ElbowPlot(experiment.aggregate)
#usually choose 9-11
```

The JackStraw function randomly permutes a subset of data, and calculates projected PCA scores for these 'random' genes, then compares the PCA scores for the 'random' genes with the observed PCA scores to determine statistical signifance. End result is a p-value for each gene's association with each principal component. We identify significant PCs as those who have a strong enrichment of low p-value genes.

```{r jacstraw, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- JackStraw(object = experiment.aggregate, dims = 50)
#Normally 50
```

```{r plot_jackstraw, warning=FALSE,error=FALSE,message=FALSE, fig.width=10, fig.height=10, warning=FALSE}
experiment.aggregate <- ScoreJackStraw(experiment.aggregate, dims = 1:50)
JackStrawPlot(object = experiment.aggregate, dims = 1:50)
#Too many PC will slow down 50 is a good number for most of samples
```

## Finally, lets save the filtered and normalized data
```{r save_rdata, warning=FALSE,error=FALSE,message=FALSE}
save(experiment.aggregate, file="pca_sample_corrected.RData")
```

## Get the next Rmd file
```{r get_next_rmd, warning=FALSE,error=FALSE,message=FALSE, eval=FALSE}
download.file("https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2021-March-Single-Cell-RNA-Seq-Analysis/master/data_analysis/scRNA_Workshop-PART5.Rmd", "scRNA_Workshop-PART5.Rmd")
```

## Session Information
```{r session_info, warning=FALSE,error=FALSE,message=FALSE}
sessionInfo()
```
