---
title: "Introduction to Single Cell RNAseq Part 2"
author: "UCD Bioinformatics Core"
output:
    html_document:
      keep_md: TRUE
---

Last Updated: March 23 2021, 5pm

# Part 2: Some QA/QC, filtering and normalization

## Load libraries
```{r libraries, warning=FALSE,error=FALSE,message=FALSE}
library(Seurat)
library(biomaRt)
library(ggplot2)
library(knitr)
library(kableExtra)
```

## Load the Seurat object from part 1
```{r load, warning=FALSE,error=FALSE,message=FALSE}
#load(file="original_seurat_object.RData")
experiment.aggregate
set.seed(12345)
```

## Some basic QA/QC of the metadata, print tables of the 5% quantiles.

Show 5% quantiles for number of genes per cell per sample
```{r quantiles_1, warning=FALSE,error=FALSE,message=FALSE}
kable(do.call("cbind", tapply(experiment.aggregate$nFeature_RNA, 
                      Idents(experiment.aggregate),quantile,probs=seq(0,1,0.05))),
      caption = "5% Quantiles of Genes/Cell by Sample") %>% kable_styling()
```

Show 5% quantiles for number of UMI per cell per sample
```{r quantiles_2, warning=FALSE,error=FALSE,message=FALSE}
kable(do.call("cbind", tapply(experiment.aggregate$nCount_RNA, 
                                      Idents(experiment.aggregate),quantile,probs=seq(0,1,0.05))),
      caption = "5% Quantiles of UMI/Cell by Sample") %>% kable_styling()
```

Show 5% quantiles for number of mitochondrial percentage per cell per sample
```{r quantiles_3, warning=FALSE,error=FALSE,message=FALSE}
kable(round(do.call("cbind", tapply(experiment.aggregate$percent.mito, Idents(experiment.aggregate),quantile,probs=seq(0,1,0.05))), digits = 3),
      caption = "5% Quantiles of Percent Mitochondria by Sample") %>% kable_styling()
```

Violin plot of 1) number of genes, 2) number of UMI and 3) percent mitochondrial genes
```{r violins, warning=FALSE,error=FALSE,message=FALSE, fig.height=12}
VlnPlot(
  experiment.aggregate,
  features = c("nFeature_RNA", "nCount_RNA","percent.mito"),
  ncol = 1, pt.size = 0.3)
```

plot ridgeplots of the same data

```{r ridgeplot_pre, warning=FALSE,error=FALSE,message=FALSE}
RidgePlot(experiment.aggregate, features=c("nFeature_RNA","nCount_RNA", "percent.mito"), ncol = 2)
```

Plot the distribution of number of cells each gene is represented by
```{r gene_range, warning=FALSE,error=FALSE,message=FALSE}
plot(sort(Matrix::rowSums(GetAssayData(experiment.aggregate) >= 3)) , xlab="gene rank", ylab="number of cells", main="Cells per genes (reads/gene >= 3 )")
```


Gene Plot, scatter plot of gene expression across cells, (colored by sample), drawing horizontal an verticale lines at proposed filtering cutoffs.


```{r relationships, warning=FALSE,error=FALSE,message=FALSE}
#FeatureScatter(experiment.aggregate, feature1 = "nCount_RNA", feature2 = "percent.mito") + geom_vline(xintercept = c(1000,12000)) + geom_hline(yintercept = 8)

#FeatureScatter(experiment.aggregate, feature1 = "nFeature_RNA", feature2 = "percent.mito") + geom_vline(xintercept = 700) + geom_hline(yintercept = 8)

FeatureScatter(
  experiment.aggregate, "nCount_RNA", "nFeature_RNA",
  pt.size = 0.5)  + geom_vline(xintercept = c(1000,12000)) + geom_hline(yintercept = 700)

```

### Cell filtering

We use the information above to filter out cells. Here we choose those that have percent mitochondrial genes max of 8%, unique UMI counts under 1,000 or greater than 12,000 and contain at least 700 features within them.

```{r}
table(experiment.aggregate$orig.ident)

#experiment.aggregate <- subset(experiment.aggregate, percent.mito <= 8)


#experiment.aggregate <- subset(experiment.aggregate, nCount_RNA >= 1000 & nCount_RNA <= 1000000)

experiment.aggregate <- subset(experiment.aggregate, nFeature_RNA >= 700)

experiment.aggregate

table(experiment.aggregate$orig.ident)
```

Lets se the ridgeplots now after filtering
```{r ridgeplot_post, warning=FALSE,error=FALSE,message=FALSE}
RidgePlot(experiment.aggregate, features=c("nFeature_RNA","nCount_RNA", "percent.mito"), ncol = 2)
```
### You may also want to filter out additional genes.

When creating the base Seurat object we did filter out some genes, recall _Keep all genes expressed in >= 10 cells_. After filtering cells and you may want to be more aggressive with the gene filter. Seurat doesn't supply such a function (that I can find), so below is a function that can do so, it filters genes requiring a min.value (log-normalized) in at least min.cells, here expression of 1 in at least 400 cells.

```{r gene_filter_fun, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate
FilterGenes <-
 function (object, min.value=1, min.cells = 0, genes = NULL) {
   genes.use <- rownames(object)
   if (!is.null(genes)) {
     genes.use <- intersect(genes.use, genes)
     object@data <- GetAssayData(object)[genes.use, ]
   } else if (min.cells > 0) {
     num.cells <- Matrix::rowSums(GetAssayData(object) > min.value)
     genes.use <- names(num.cells[which(num.cells >= min.cells)])
     object = object[genes.use, ]
   }
  object <- LogSeuratCommand(object = object)
  return(object)
}

experiment.aggregate.genes <- FilterGenes(object = experiment.aggregate, min.value = 1, min.cells = 400)
experiment.aggregate.genes
rm(experiment.aggregate.genes)
```

## FOR THIS WORKSHOP ONLY: Lets reduce the dataset down to 1000 cells per sample for time
```{r reduce dataset, warning=FALSE,error=FALSE,message=FALSE}
table(Idents(experiment.aggregate))
#experiment.aggregate <- experiment.aggregate[,unlist(sapply(split(seq_along(Cells(experiment.aggregate)), experiment.aggregate$orig.ident), sample, size=1000, simplify = F))]
table(Idents(experiment.aggregate))
```

## Next we want to normalize the data

After filtering out cells from the dataset, the next step is to normalize the data. By default, we employ a global-scaling normalization method LogNormalize that normalizes the gene expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and then log-transforms the data.

```{r normalize_help, warning=FALSE,error=FALSE,message=FALSE, eval=FALSE}
?NormalizeData
```

```{r normalize, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- NormalizeData(
  object = experiment.aggregate,
  normalization.method = "LogNormalize",
  scale.factor = 10000)
```

### Calculate Cell-Cycle with Seurat, the list of genes comes with Seurat (only for human)
[Dissecting the multicellular ecosystem of metastatic melanoma by single-cell RNA-seq](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4944528/)

First need to convert to mouse symbols, we'll use Biomart for that too.
```{r cellcycle, warning=FALSE,error=FALSE,message=FALSE}
#Mouse Code
 convertHumanGeneList <- function(x){
   require("biomaRt")
   human = useEnsembl("ensembl", dataset = "hsapiens_gene_ensembl", mirror = "uswest")
   mouse = useEnsembl("ensembl", dataset = "mmusculus_gene_ensembl", mirror = "uswest")
 
   genes = getLDS(attributes = c("hgnc_symbol"), filters = "hgnc_symbol", values = x , mart = human, attributesL = c("mgi_symbol"), martL = mouse, uniqueRows=T)
 
   humanx <- unique(genes[, 2])
 
   # Print the first 6 genes found to the screen
   print(head(humanx))
   return(humanx)
 }
# 
 m.s.genes <- convertHumanGeneList(cc.genes.updated.2019$s.genes)
 m.g2m.genes <- convertHumanGeneList(cc.genes.updated.2019$g2m.genes)
# 
# # Create our Seurat object and complete the initialization steps
# experiment.aggregate <- CellCycleScoring(experiment.aggregate, s.features = m.s.genes, g2m.features = m.g2m.genes, set.ident = TRUE)

# Human Code
#s.genes <- (cc.genes$s.genes)
#g2m.genes <- (cc.genes$g2m.genes)

# Create our Seurat object and complete the initialization steps
experiment.aggregate <- CellCycleScoring(experiment.aggregate, s.features = m.s.genes, g2m.features = m.g2m.genes, set.ident = TRUE)
```

#### Table of cell cycle (seurate)

```{r cellcycle_table, warning=FALSE,error=FALSE,message=FALSE}
table(experiment.aggregate@meta.data$Phase) %>% kable(caption = "Number of Cells in each Cell Cycle Stage", col.names = c("Stage", "Count"), align = "c") %>% kable_styling()
```

#### Fixing the defualt "Ident" in Seurat

```{r update_ident, warning=FALSE,error=FALSE,message=FALSE}
table(Idents(experiment.aggregate))
## So lets change it back to samplename
Idents(experiment.aggregate) <- "orig.ident"
table(Idents(experiment.aggregate))
```


## Identify variable genes

The function FindVariableFeatures identifies the most highly variable genes (default 2000 genes) by fitting a line to the relationship of log(variance) and log(mean) using loess smoothing, uses this information to standardize the data, then calculates the variance of the standardized data.  This helps avoid selecting genes that only appear variable due to their expression level.

```{r fine_variable_genes_help, warning=FALSE,error=FALSE, message=FALSE, eval=FALSE}
?FindVariableFeatures
```

```{r find_variable_genes, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate_filtered <-experiment.aggregate
experiment.aggregate <- FindVariableFeatures(
  object = experiment.aggregate,
  selection.method = "vst")

length(VariableFeatures(experiment.aggregate))

top10 <- head(VariableFeatures(experiment.aggregate), 10)

top10

vfp1 <- VariableFeaturePlot(experiment.aggregate)
vfp1 <- LabelPoints(plot = vfp1, points = top10, repel = TRUE)
vfp1
```

#### Question(s)

1. Play some with the filtering parameters, see how results change?
2. How do the results change if you use selection.method = "dispersion" or selection.method = "mean.var.plot"

```{r find_variable_genes, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- FindVariableFeatures(
  object = experiment.aggregate_filtered,
  selection.method = "dispersion")


length(VariableFeatures(experiment.aggregate))

top10 <- head(VariableFeatures(experiment.aggregate), 10)

top10

dispersion1 <- VariableFeaturePlot(experiment.aggregate)
dispersion1 <- LabelPoints(plot = dispersion1, points = top10, repel = TRUE)
dispersion1

```

```{r find_variable_genes, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- FindVariableFeatures(
  object = experiment.aggregate_filtered,
  selection.method = "mean.var.plot")

length(VariableFeatures(experiment.aggregate))


top10 <- head(VariableFeatures(experiment.aggregate), 10)

top10

meanvar1 <- VariableFeaturePlot(experiment.aggregate)
meanvar1 <- LabelPoints(plot = meanvar1, points = top10, repel = TRUE)
meanvar1

```

## Finally, lets save the filtered and normalized data
```{r save_r, warning=FALSE,error=FALSE,message=FALSE}
save(experiment.aggregate, file="pre_sample_corrected.RData")
```

## Get the next Rmd file
```{r retreive_rmd, warning=FALSE,error=FALSE,message=FALSE, eval=FALSE}
download.file("https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2021-March-Single-Cell-RNA-Seq-Analysis/master/data_analysis/scRNA_Workshop-PART3.Rmd", "scRNA_Workshop-PART3.Rmd")
```

## Session Information
```{r session_info, warning=FALSE,error=FALSE,message=FALSE}
sessionInfo()
```
