---
title: "Introduction to Single Cell RNAseq Part 5"
author: "UCD Bioinformatics Core"
output:
    html_document:
      keep_md: TRUE
---

Last Updated: March 24 2021, 5pm

# Part 5: 

## Load libraries
```{r load_libraries, warning=FALSE,error=FALSE,message=FALSE}
library(Seurat)
library(ggplot2)
library(dplyr)
```

## Load the Seurat object
```{r load_rdata, warning=FALSE,error=FALSE,message=FALSE}
load(file="pca_sample_corrected.RData")
experiment.aggregate
```


## So how many features should we use? Use too few and your leaving out interesting variation that may define cell types, use too many and you add in noise? maybe?

Lets choose the first 50, based on our prior part.

```{r use_pcs, warning=FALSE,error=FALSE,message=FALSE}
use.pcs = 1:50
```

## Identifying clusters

Seurat implements an graph-based clustering approach. Distances between the cells are calculated based on previously identified PCs. 

The default method for identifying k-nearest neighbors has been changed in V4 to [annoy](https://github.com/spotify/annoy) ("Approximate Nearest Neighbors Oh Yeah!). This is an approximate nearest-neighbor approach that is widely used for high-dimensional analysis in many fields, including single-cell analysis. Extensive community benchmarking has shown that annoy substantially improves the speed and memory requirements of neighbor discovery, with negligible impact to downstream results. 



Seurat prior approach was heavily inspired by recent manuscripts which applied graph-based clustering approaches to scRNAseq data. Briefly, Seurat identified clusters of cells by a shared nearest neighbor (SNN) modularity optimization based clustering algorithm. First calculate k-nearest neighbors (KNN) and construct the SNN graph. Then optimize the modularity function to determine clusters. For a full description of the algorithms, see Waltman and van Eck (2013) The European Physical Journal B. You can switch back to using the previous default setting using nn.method="rann".


The FindClusters function implements the neighbor based clustering procedure, and contains a resolution parameter that sets the granularity of the downstream clustering, with increased values leading to a greater number of clusters. I tend to like to perform a series of resolutions, investigate and choose.

```{r find_neighbors_help, warning=FALSE,error=FALSE,message=FALSE, eval=FALSE}
?FindNeighbors
```

```{r find_neighbors, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- FindNeighbors(experiment.aggregate, reduction="pca", dims = use.pcs)

experiment.aggregate <- FindClusters(
    object = experiment.aggregate,
    resolution = seq(0.25,4,0.5), #top/bottom of tree
    #resolution = seq(0.1,4,0.1), #top/bottom of tree
    verbose = FALSE
)
#Partially Supervised clustering
#Resolution: you need to decide(smaller numbers: less cluster, large numbers: more cluster)

```


Seurat add the clustering information to the metadata beginning with RNA_snn_res. followed by the resolution

```{r cluster_metadata, warning=FALSE,error=FALSE,message=FALSE}
head(experiment.aggregate[[]])
```


Lets first investigate how many clusters each resolution produces and set it to the smallest resolutions of 0.5 (fewest clusters).

```{r clusters_per_res, warning=FALSE,error=FALSE,message=FALSE}
sapply(grep("res",colnames(experiment.aggregate@meta.data),value = TRUE),
       function(x) length(unique(experiment.aggregate@meta.data[,x])))
```

### Plot TSNE coloring for each resolution

tSNE dimensionality reduction plots are then used to visualize clustering results. As input to the tSNE, you should use the same PCs as input to the clustering analysis.

```{r create_tsne, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- RunTSNE(
  object = experiment.aggregate,
  reduction.use = "pca",
  dims = use.pcs,
  do.fast = TRUE)
```


```{r tsne_all_1, warning=FALSE,error=FALSE,message=FALSE, fig.height=10}
DimPlot(object = experiment.aggregate, group.by=grep("res",colnames(experiment.aggregate@meta.data),value = TRUE)[1:4], ncol=2 , pt.size=3.0, reduction = "tsne",label.size = 10, label = T)
```

```{r tsne_all_2, warning=FALSE,error=FALSE,message=FALSE, fig.height=10}
DimPlot(object = experiment.aggregate, group.by=grep("res",colnames(experiment.aggregate@meta.data),value = TRUE)[5:8], ncol=2 , pt.size=3.0, reduction = "tsne", label = T)
```

1. Try exploring different PCS, so first 5, 15, 25, we used 50, what about 100? How does the clustering change?
#100 is not useful

Once complete go back to 1:50

### Choosing a resolution

Lets set the default identity to a resolution of 0.25 and produce a table of cluster to sample assignments.
```{r cells_per_res_sample, warning=FALSE,error=FALSE,message=FALSE}
Idents(experiment.aggregate) <- "RNA_snn_res.1.25"
table(Idents(experiment.aggregate),experiment.aggregate$orig.ident)
```

Plot TSNE coloring by the slot 'ident' (default).
```{r plot_tsne, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate, pt.size=0.5, reduction = "tsne", label = T)
```


### uMAP dimensionality reduction plot.

```{r umap, warning=FALSE,error=FALSE,message=FALSE}
experiment.aggregate <- RunUMAP(
  object = experiment.aggregate,
  dims = use.pcs)
```

Plot uMap coloring by the slot 'ident' (default).
```{r plot_umap, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate, pt.size=0.5, reduction = "umap", label = T)
```

Catagorical data can be plotted using the DimPlot function.


TSNE plot by cell cycle
```{r plot_cellcycle, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate, pt.size=0.5, group.by = "Phase", reduction = "umap" )
```

1. Try creating a table, of cluster x cell cycle

### Can use feature plot to plot our read valued metadata, like nUMI, Feature count, and percent Mito
FeaturePlot can be used to color cells with a 'feature', non categorical data, like number of UMIs
```{r plot_rna, warning=FALSE,error=FALSE,message=FALSE}
FeaturePlot(experiment.aggregate, features = c('nCount_RNA'), pt.size=0.5)
```
and number of genes present
```{r plot_feature, warning=FALSE,error=FALSE,message=FALSE}
FeaturePlot(experiment.aggregate, features = c('nFeature_RNA'), pt.size=0.5)
```

percent mitochondrial
```{r plot_mito, warning=FALSE,error=FALSE,message=FALSE}
FeaturePlot(experiment.aggregate, features = c('percent.mito'), pt.size=0.5)
```

## Building a phylogenetic tree relating the 'average' cell from each group in default 'Ident' (currently "RNA_snn_res.0.25"). Tree is estimated based on a distance matrix constructed in either gene expression space or PCA space.

```{r create_tree, warning=FALSE,error=FALSE,message=FALSE}
library(ape)
#Idents(experiment.aggregate)=experiment.aggregate$orig.ident
experiment.aggregate <- BuildClusterTree(
  experiment.aggregate, dims = use.pcs)
#experiment.aggregate <- BuildClusterTree(features = 'nCount_RNA',
#  experiment.aggregate, dims = use.pcs)



PlotClusterTree(experiment.aggregate)
```
1. Create new trees of other data

Once complete go back to Res 0.25

```{r umap_plot2, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate, pt.size=0.5, label = TRUE, reduction = "umap")
```

### Merging clusters

Merge Clustering results, so lets say clusters 3 and 7 are actually the same cell type (overlaps in the tsne, not as apparent in the umap) and we don't wish to separate them out as distinct clusters. Same with 4 and 5.

```{r merging_cluster, warning=FALSE,error=FALSE,message=FALSE}
experiment.merged = experiment.aggregate
Idents(experiment.merged) <- "RNA_snn_res.0.75"

#experiment.merged <- RenameIdents(
#  object = experiment.merged,
#  '2' = '4' ## combining clusters
#)

table(Idents(experiment.merged))
DimPlot(object = experiment.merged, pt.size=0.5, label = T, reduction = "umap")
VlnPlot(object = experiment.merged, features = "percent.mito", pt.size = 0.05)
```

### Reording the clusters

In order to reorder the clusters for plotting purposes take a look at the levels of the Ident, which indicates the ordering, then relevel as desired.

```{r merging_cluster2, warning=FALSE,error=FALSE,message=FALSE}
experiment.examples <- experiment.merged
levels(experiment.examples@active.ident)

experiment.examples@active.ident <- relevel(experiment.examples@active.ident, "2")
levels(experiment.examples@active.ident)
# now cluster 6 is the "first" factor

DimPlot(object = experiment.examples, pt.size=0.5, label = T, reduction = "umap")
VlnPlot(object = experiment.examples, features = "percent.mito", pt.size = 0.05)
```


```{r merging_cluster3, warning=FALSE,error=FALSE,message=FALSE}
# relevel all the factors to the order I want
Idents(experiment.examples) <- factor(experiment.examples@active.ident, levels=c("6","2","0","7","5","9","1","8"))
levels(experiment.examples@active.ident)

DimPlot(object = experiment.examples, pt.size=0.5, label = T, reduction = "umap")
```


### Re-assign clustering result (subclustering only cluster 0) to clustering for resolution 3.75  (@ reslution 0.25) [adding a R prefix]
```{r subclusters, warning=FALSE,error=FALSE,message=FALSE}
newIdent = as.character(Idents(experiment.examples))
newIdent[newIdent == '0'] = paste0("R",as.character(experiment.examples$RNA_snn_res.3.75[newIdent == '0'])) #using cluster 0 cells and see any sub clusters in high resolution

Idents(experiment.examples) <- as.factor(newIdent)
table(Idents(experiment.examples))
```

```{r subclusters_plot, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.examples, pt.size=0.5, label = T, reduction = "umap")
```

Plot UMAP  coloring by the slot 'orig.ident' (sample names) with alpha colors turned on. A pretty picture
```{r pretty_pre, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate, group.by="orig.ident", pt.size=0.5, reduction = "umap" )
```


```{r pretty_post, warning=FALSE,error=FALSE,message=FALSE}
## Pretty umap using alpha
alpha.use <- 2/5
p <- DimPlot(object = experiment.aggregate, group.by="orig.ident", pt.size=0.5, reduction = "umap")
p$layers[[1]]$mapping$alpha <- alpha.use
p + scale_alpha_continuous(range = alpha.use, guide = F)

```

Removing cells assigned to clusters from a plot, So here plot all clusters but cluster 6 (contaminant?)
```{r removing_cells, warning=FALSE,error=FALSE,message=FALSE}
# create a new tmp object with those removed
experiment.aggregate.tmp <- experiment.aggregate[,-which(Idents(experiment.aggregate) %in% c("1"))]

dim(experiment.aggregate)
dim(experiment.aggregate.tmp)
```


```{r removing_cells_plot, warning=FALSE,error=FALSE,message=FALSE}
DimPlot(object = experiment.aggregate.tmp, group.by="orig.ident", pt.size=0.5, reduction = "umap", label = T)
```

## Identifying Marker Genes

Seurat can help you find markers that define clusters via differential expression.

`FindMarkers` identifies markers for a cluster relative to all other clusters.

`FindAllMarkers` does so for all clusters

`FindAllMarkersNode` defines all markers that split a Node from the cluster tree

```{r find_markers_help, warning=FALSE,error=FALSE,message=FALSE}
?FindMarkers
```

```{r find_markers, warning=FALSE,error=FALSE,message=FALSE}
#markers = FindMarkers(experiment.merged, ident.1=c(6))
#markers = FindMarkers(experiment.merged, ident.1=c(2,7),ident.2=c(1,5)) #using 2=differential expression
experiment.merged@active.ident
markers = FindMarkers(experiment.merged, ident.1=0,ident.2=1)
#use node number: findallmarkersnode
#markers2 = FindAllMarkers(experiment.merged)

library(biomaRt)
ensembl <- useEnsembl(biomart = "genes", dataset = "mmusculus_gene_ensembl")
#listEnsembl()
MyensembleGenes_orig=rownames(markers)
MyensembleGenes <- sapply(strsplit(MyensembleGenes_orig,"\\."), function(x) x[1])
HUGOID = getBM(attributes = c('ensembl_gene_id', 'external_gene_name'), 
              #filters = 'ensembl_gene_id', 
              values = MyensembleGenes, 
              mart = ensembl)
Mygenescombined=cbind(MyensembleGenes_orig,MyensembleGenes)
HUGOIDMATCHING=HUGOID[match(Mygenescombined[,2],HUGOID[,1]),]
Mygenescombined=cbind(Mygenescombined,HUGOIDMATCHING)
markersgenename=cbind(markers,HUGOIDMATCHING)

head(markersgenename)
dim(markersgenename)
table(markers$avg_log2FC > 0) #log2 foldchange for new version
table(markers$p_val_adj < 0.05)

```


pct.1 and pct.2 are the proportion of cells with expression above 0 in ident.1 and ident.2 respectively. p_val is the raw p_value associated with the differntial expression test with adjusted value in p_val_adj. avg_logFC is the average log fold change difference between the two groups.

avg_diff (lines 130, 193 and) appears to be the difference in log(x = mean(x = exp(x = x) - 1) + 1) between groups.  It doesn’t seem like this should work out to be the signed ratio of pct.1 to pct.2 so I must be missing something.  It doesn’t seem to be related at all to how the p-values are calculated so maybe it doesn’t matter so much, and the sign is probably going to be pretty robust to how expression is measured.

Can use a violin plot to visualize the expression pattern of some markers
```{r vln, warning=FALSE,error=FALSE,message=FALSE}
VlnPlot(object = experiment.aggregate, features = rownames(markers), pt.size = 0.05)
```

Or a feature plot
```{r gene_feature, warning=FALSE,error=FALSE,message=FALSE}
FeaturePlot(
    experiment.merged,
    head(rownames(markers), n=2),
    cols = c("lightgrey", "blue"),
    ncol = 2
)
```

FindAllMarkers can be used to automate the process across all genes.

```{r find_all_markers, warning=FALSE,error=FALSE,message=FALSE}
markers_all <- FindAllMarkers(
    object = experiment.merged,
    only.pos = TRUE,
    min.pct = 0.25,
    thresh.use = 0.25
)
dim(markers_all)
head(markers_all)
table(table(markers_all$gene))

markers_all_single <- markers_all[markers_all$gene %in% names(table(markers_all$gene))[table(markers_all$gene) == 1],]

dim(markers_all_single)
table(table(markers_all_single$gene))

table(markers_all_single$cluster)

head(markers_all_single)
```



Plot a heatmap of genes by cluster for the top 10 marker genes per cluster
```{r markers_head, warning=FALSE,error=FALSE,message=FALSE, fig.height=7}
top10 <- markers_all_single %>% group_by(cluster) %>% top_n(10, avg_log2FC)
DoHeatmap(
    object = experiment.merged,
    features = top10$gene
)
```

```{r get_means, warning=FALSE,error=FALSE,message=FALSE}
# Get expression of genes for cells in and out of each cluster
getGeneClusterMeans <- function(gene, cluster){
  x <- GetAssayData(experiment.merged)[gene,]
  m <- tapply(x, ifelse(Idents(experiment.merged) == cluster, 1, 0), mean)
  mean.in.cluster <- m[2]
  mean.out.of.cluster <- m[1]
  return(list(mean.in.cluster = mean.in.cluster, mean.out.of.cluster = mean.out.of.cluster))
}

## for sake of time only using first six (head)
means <- mapply(getGeneClusterMeans, head(markers_all[,"gene"]), head(markers_all[,"cluster"]))
means <- matrix(unlist(means), ncol = 2, byrow = T)

colnames(means) <- c("mean.in.cluster", "mean.out.of.cluster")
rownames(means) <- head(markers_all[,"gene"])
markers_all2 <- cbind(head(markers_all), means)
head(markers_all2)
```

## Finishing up clusters.

At this point in time you should use the tree, markers, domain knowledge, and goals to finalize your clusters. This may mean adjusting PCA to use, mergers clusters together, choosing a new resolutions, etc. When finished you can further name it cluster by something more informative. Ex.
```{r finish_cluster, warning=FALSE,error=FALSE,message=FALSE}
experiment.clusters <- experiment.aggregate
experiment.clusters <- RenameIdents(
  object = experiment.clusters,
  '0' = 'cell_type_A',
  '1' = 'cell_type_B',
  '2' = 'cell_type_C',
  '3' = 'cell_type_D',
  '4' = 'cell_type_E'
)
# and so on

DimPlot(object = experiment.clusters, pt.size=0.5, label = T, reduction = "umap")
```

Right now our results ONLY exist in the Ident data object, lets save it to our metadata table so we don't accidentally loose it.
```{r save_cluster, warning=FALSE,error=FALSE,message=FALSE}
experiment.merged$finalcluster <- Idents(experiment.merged)
head(experiment.merged[[]])
#FeaturePlot(object = experiment.merged, features.plot = c("CD3D"), min.cutoff = "q9", cols.use = c("lightgrey", "blue"), pt.size = 0.5)

```

## Subsetting samples and plotting

If you want to look at the representation of just one sample, or sets of samples
```{r subset, warning=FALSE,error=FALSE,message=FALSE}
experiment.sample2 <- subset(experiment.merged, orig.ident == "YES")

DimPlot(object = experiment.sample2, group.by = "RNA_snn_res.0.75", pt.size=0.5, label = TRUE, reduction = "tsne")
experiment.sample1 <- subset(experiment.merged, orig.ident == "NO")

DimPlot(object = experiment.sample1, group.by = "RNA_snn_res.0.75", pt.size=0.5, label = TRUE, reduction = "tsne")


#experiment.batch1 <- subset(experiment.merged, batchid == "6")

#DimPlot(object = experiment.batch1, group.by = "RNA_snn_res.0.25", pt.size=0.5, label = TRUE, reduction = "tsne")
```

### Adding in a new metadata column representing samples within clusters. So differential expression of PBMC2 vs PBMC3 within cluster 7

```{r, warning=FALSE,error=FALSE,message=FALSE, fig.height=7}
experiment.merged$samplecluster = paste(experiment.merged$orig.ident,experiment.merged$finalcluster,sep = '-')

# set the identity to the new variable
Idents(experiment.merged) <- "samplecluster"

markers.comp0 <- FindMarkers(experiment.merged, ident.1 = c("YES-0"), ident.2= c("NO-0"))
markers.comp1 <- FindMarkers(experiment.merged, ident.1 = c("YES-1"), ident.2= c("NO-1"))
markers.comp2 <- FindMarkers(experiment.merged, ident.1 = c("YES-2"), ident.2= c("NO-2"))
markers.comp3 <- FindMarkers(experiment.merged, ident.1 = c("YES-3"), ident.2= c("NO-3"))
markers.group <- FindMarkers(experiment.merged, ident.1=c("YES-0"), ident.2=c("YES-1"))
head(markers.comp0)
markers.comp0.ordered=markers.comp0[order(markers.comp0$p_val_adj, decreasing = FALSE, na.last = TRUE, method = c("auto")),] 

experiment.subset <- subset(experiment.merged, samplecluster %in%  c( "YES-0", "YES-1" ))
experiment.subset.comp0 <- subset(experiment.merged, samplecluster %in%  c( "YES-0", "NO-0" ))
experiment.subset.comp1 <- subset(experiment.merged, samplecluster %in%  c( "YES-1", "NO-1" ))
experiment.subset.comp2 <- subset(experiment.merged, samplecluster %in%  c( "YES-2", "NO-2" ))
experiment.subset.comp3 <- subset(experiment.merged, samplecluster %in%  c( "YES-3", "NO-3" ))


DoHeatmap(experiment.subset, features = head(rownames(markers.group),20))
DoHeatmap(experiment.subset.comp0, features = head(rownames(markers.comp0),20))
DoHeatmap(experiment.subset.comp1, features = head(rownames(markers.comp1),20))
DoHeatmap(experiment.subset.comp2, features = head(rownames(markers.comp2),20))
DoHeatmap(experiment.subset.comp3, features = head(rownames(markers.comp3),20))


#listEnsembl()
MyensembleGenes_orig=rownames(markers.comp0)
MyensembleGenes = sapply(strsplit(rownames(markers.comp0),"\\."), function(x) x[1])
symb <- mapIds(org.Mm.eg.db, keys=MyensembleGenes, keytype="ENSEMBL", column="SYMBOL")
rownames(markers.comp0) <- uniquifyFeatureNames(rownames(markers.comp0), symb)

DoHeatmap(experiment.subset, features = head(rownames(markers.comp0),20))


```


```{r change_back_to_final, warning=FALSE,error=FALSE,message=FALSE}
Idents(experiment.merged) <- "finalcluster"
```

And last lets save all the objects in our session.
```{r save_rdata, warning=FALSE,error=FALSE,message=FALSE}
save(list=ls(), file="clusters_seurat_object.RData")
```

## Get the next Rmd file
```{r get_next_rmd, warning=FALSE,error=FALSE,message=FALSE, eval=FALSE}
download.file("https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2021-March-Single-Cell-RNA-Seq-Analysis/master/data_analysis/scRNA_Workshop-PART6.Rmd", "scRNA_Workshop-PART6.Rmd")
```

## Session Information
```{r session_info, warning=FALSE,error=FALSE,message=FALSE}
sessionInfo()
```
